### PR TITLE
addendum: Add name tag to spark3.0.1 notebook image for it to show

### PR DESCRIPTION
**Pull request by oshritf**

### CODE REVIEW

**Comment by HumairAK** [comment]
thanks! can you rebase/squash these to 1 commit?

**Comment by oshritf** [comment]
@HumairAK done

**Review by HumairAK — APPROVED** [review]
/lgtm
/approve

**Comment by HumairAK** [comment]
/hold
trying this out on stage environment, coming into some issues, will report back

File: kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml [hunk 1]
```diff
@@ -18,3 +18,4 @@ spec:
     from:
       kind: DockerImage
       name: quay.io/llasmith/s2i-spark-container:py36-spark3.0.1-hadoop3.2.0
```
Comment (4n4nd) [inline kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml:20]: I was tested this image in the staging env and I found that the `ImageStream` name should be the same name as the image. So [here in line 11](https://github.com/operate-first/apps/blob/0c50a36d273140972d9bf069be1fa53018dd8ca4/kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml#L11), the name should be `s2i-spark-container`

**Comment by 4n4nd** [comment]
Also, when I tried to run this image, I see the following error in pod logs
```
/usr/bin/container-entrypoint: line 2: exec: start-singleuser.sh: not found
```

this was with the following image: 
`quay.io/llasmith/s2i-spark-container@sha256:0e850489252e6caec6a3029a87af3b0dce826404353b6f0279932e2eb0d2059b`

**Review by 4n4nd — CHANGES_REQUESTED** [review]
requested changes in above comments

File: kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml (hunk 1 above)
Comment (4n4nd) [reply kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml:20]: If you don't do this, we see the following error:
`Back-off pulling image "spark3.0.1-xskipper:py36-spark3.0.1-hadoop3.2.0"`

Comment (4n4nd) [reply kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml:20]: @vpavlin @LaVLas is this the desired behavior?

**Comment by oshritf** [comment]
> Also, when I tried to run this image, I see the following error in pod logs
> 
> ```
> /usr/bin/container-entrypoint: line 2: exec: start-singleuser.sh: not found
> ```
> 
> this was with the following image:
> `quay.io/llasmith/s2i-spark-container@sha256:0e850489252e6caec6a3029a87af3b0dce826404353b6f0279932e2eb0d2059b`

@anishasthana

File: kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml (hunk 1 above)
Comment (LaVLaS) [reply kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml:20]: No. I created that image as a test to see if I could recreate the images that @oshritf experienced when building docker locally.  I never tested this with an actual notebook build so it should not be used in an official deployment

We do have a spark version update on the ODH roadmap but this particular image would not be used for that work.

Comment (4n4nd) [reply kfdefs/base/jupyterhub/notebook-images/s2i-spark3.0.1-minimal-notebook.yaml:20]: My question was more about the need for ImageStream name and Image name to be the same for JH to use the correct image.

**Pull request closed without merging**